File: client/java/src/avt/InputDlg.java
package avt;

import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;

public final class InputDlg extends Dialog {
   private String[] info;
   private TField tfInput = new TField();
   private IAction okAction;
   private Image img;
   private int w;
   private int h;

   public InputDlg() {
      this.tfInput.e = false;
   }

   public final void commandTab(int var1, int var2) {
      switch (var1) {
         case 120:
            Canvas.currentDialog = null;
            return;
         default:
            Canvas.currentMyScreen.commandTab(var1, var2);
      }
   }

   public final void setImg(Image var1) {
      this.img = var1;
      this.h += var1.getHeight();
      this.init();
   }

   public final String getText() {
      return this.tfInput.getText();
   }

   public final void init() {
      this.tfInput.x = Canvas.hw - this.tfInput.width / 2;
      this.tfInput.y = Canvas.h - (Canvas.h - Canvas.ae[0].y + 5) - this.tfInput.height - 8;
   }

   public final void setInfoIkb(String var1, int var2, int var3) {
      this.initInfo(var1, var3);
      super.center = new Command(T.OK, var2);
      Canvas.currentDialog = this;
      this.tfInput.setFocus(true);
   }

   private void initInfo(String var1, int var2) {
      this.img = null;
      this.w = Canvas.w - 40;
      this.h = 70 * AvMain.hd;
      if (Canvas.normalFont.getWidth(var1) + 20 < this.w) {
         this.w = Canvas.normalFont.getWidth(var1) + 20;
      }

      if (this.w < Canvas.w / 2) {
         this.w = Canvas.w / 2;
      }

      this.info = Canvas.normalFont.splitFontBStrInLine(var1, this.w - 20);
      this.tfInput = new TField();
      this.tfInput.e = false;
      this.tfInput.width = this.w - 10;
      this.init();
      this.tfInput.setText("");
      this.tfInput.setIputType(var2);
      super.left = new Command(T.close, 120);
      Canvas.currentDialog = this;
   }

   public final void setImg(String var1, IAction var2, int var3) {
      this.initInfo(var1, var3);
      this.okAction = var2;
      super.center = new Command(T.OK, this.okAction);
      Canvas.currentDialog = this;
   }

   public final void setImg(String var1, IAction var2, int var3, String var4) {
      this.initInfo(var1, var3);
      if (var4 != null) {
         this.tfInput.setText(var4);
      }

      this.okAction = var2;
      super.center = new Command(T.OK, this.okAction);
      Canvas.currentDialog = this;
   }

   public final void paint(Graphics var1) {
      Canvas.resetTrans(var1);
      Canvas.paint.paintPopupBack(var1, Canvas.hw - this.w / 2, Canvas.h - this.h - (Canvas.h - Canvas.ae[0].y + 5), this.w, this.h, 0);
      int var2 = Canvas.h - this.h - (Canvas.h - Canvas.ae[0].y + 5) + (this.h - this.tfInput.height - 8) / 2 - (this.info.length >> 1) * AvMain.hNormal - AvMain.hNormal / 2;
      if (this.img != null) {
         var1.drawImage(this.img, Canvas.hw, this.tfInput.y - this.img.getHeight() / 2 - 5 * AvMain.hd, 3);
         var2 -= this.img.getHeight() / 2;
      }

      int var3 = 0;

      for(var2 = var2; var3 < this.info.length; var2 += AvMain.hNormal) {
         Canvas.normalFont.drawString(var1, this.info[var3], Canvas.hw, var2, 2);
         ++var3;
      }

      this.tfInput.paint(var1);
      if (OnScreen.isOngame) {
         Canvas.resetTrans(var1);
         Canvas.paint.paintBackground(var1);
         Canvas.paint.paintCommandAlt(var1, super.left, super.center, super.right);
      } else {
         super.paint(var1);
      }

   }

   public final void keyPress(int var1) {
      this.tfInput.keyPressed(var1);
   }

   public final void updateKey() {
      this.tfInput.update();
      if (this.tfInput.isFocused()) {
         super.right = this.tfInput.getRightCmd();
      }

      if (OnScreen.isOngame && Canvas.stypeInt != 0) {
         Canvas.paint.updateKeyOn(super.left, super.center, super.right);
      } else {
         super.updateKey();
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/InputDlg.cs
public class InputDlg : Dialog
{
	private class IActionOk : IAction
	{
		public void perform()
		{
			Canvas.inputDlg.center.perform();
		}
	}

	protected string[] info;

	public TField tfInput;

	public IAction okAction;

	private Image img;

	private int x;

	private int y;

	private int w;

	private int h;

	public InputDlg()
	{
		tfInput = new TField(string.Empty, null, new IActionOk());
		tfInput.isChangeFocus = false;
		tfInput.showSubTextField = false;
		tfInput.autoScaleScreen = true;
	}

	public override void commandTab(int index)
	{
		Canvas.currentMyScreen.commandTab(index);
	}

	public string getText()
	{
		return tfInput.getText();
	}

	public override void init(int hCan)
	{
		y = hCan - h - 50;
		tfInput.x = Canvas.hw - tfInput.width / 2;
		tfInput.y = y + h - tfInput.height - AvMain.hCmd / 2 - 10;
		if (center != null)
		{
			center.x = Canvas.w / 2;
			center.y = y + h - PaintPopup.hButtonSmall / 2;
		}
	}

	public void setImg(Image img)
	{
		this.img = img;
		h += img.getHeight();
		init(Canvas.hCan);
	}

	public void setInfo(string info, int index, int type, MyScreen parent)
	{
		initInfo(info, type);
		show();
		tfInput.parent = parent;
		tfInput.setFocus(true);
		tfInput.showSubTextField = false;
		center = new Command(T.OK, index, Canvas.hw, y + h - AvMain.hCmd / 2);
	}

	public void initInfo(string info, int type)
	{
		img = null;
		w = Canvas.w - 40;
		if (Canvas.normalFont.getWidth(info) + 20 < w)
		{
			w = Canvas.normalFont.getWidth(info) + 20;
		}
		if (w < Canvas.w / 2)
		{
			w = Canvas.w / 2;
		}
		this.info = Canvas.normalFont.splitFontBStrInLine(info, w - 20);
		h = AvMain.hCmd / 2 + 10 + tfInput.height + this.info.Length * AvMain.hNormal + 60;
		x = (Canvas.w - w) / 2;
		y = Canvas.hCan - h - 50;
		if (center != null)
		{
			center.x = Canvas.w / 2;
			center.y = y + h - PaintPopup.hButtonSmall / 2;
		}
		tfInput.isChangeFocus = false;
		tfInput.width = w - 20 * AvMain.hd;
		init(Canvas.hCan);
		tfInput.setText(string.Empty);
		tfInput.setIputType(ipKeyboard.TEXT);
		show();
	}

	public void setInfoIA(string info, IAction ok, int type, MyScreen me)
	{
		initInfo(info, type);
		okAction = ok;
		tfInput.parent = me;
		center = new Command(T.OK, okAction);
		tfInput.showSubTextField = false;
		show();
	}

	public void setInfoIkb(string info, IKbAction ac, int type, MyScreen me)
	{
		initInfo(info, type);
		tfInput.parent = me;
		center = new Command(T.OK, ac);
		init(Canvas.hCan);
		tfInput.showSubTextField = false;
		show();
	}

	public override void paint(MyGraphics g)
	{
		Canvas.resetTrans(g);
		Canvas.paint.paintPopupBack(g, x, y, w, h, -1, false);
		int num = y + (tfInput.y - y) / 2 - info.Length * AvMain.hNormal / 2 + 4 * AvMain.hd;
		if (img != null)
		{
			g.drawImage(img, x + w / 2, tfInput.y - img.getHeight() / 2 - 5 * AvMain.hd, 3);
			num -= img.getHeight() / 2;
		}
		int num2 = 0;
		int num3 = num;
		while (num2 < info.Length)
		{
			Canvas.normalFont.drawString(g, info[num2], Canvas.hw, num3, 2);
			num2++;
			num3 += AvMain.hNormal;
		}
		tfInput.paint(g);
		base.paint(g);
	}

	public override void keyPress(int keyCode)
	{
		tfInput.keyPressed(keyCode);
		if (keyCode == -5)
		{
			tfInput.action.perform();
		}
	}

	public override void updateKey()
	{
		tfInput.update();
		if (tfInput.isFocused())
		{
			right = null;
		}
		base.updateKey();
	}

	public override void show()
	{
		Canvas.currentDialog = this;
	}

	public void showWithCaption(string text)
	{
		tfInput.name = text;
		show();
	}
}
